### PR TITLE
Feature/php version bump

### PR DESCRIPTION
**Pull request by corentin-larose**
Not really a feature:

- array() => []
- CS fixes
- PHP version bump in Composer
- Added Author in Composer

### CODE REVIEW

File: src/HttpCacheListener.php [hunk 1]
```diff
@@ -19,21 +19,21 @@ class HttpCacheListener extends AbstractListenerAggregate
...
      * @var array
      */
-    protected $config = array();
+    protected $config = [];
 
     /**
      * @param EventManagerInterface $events
      * @param int                   $priority
      */
-    public function attach(EventManagerInterface $events, $priority = 1)
```
Comment (weierophinney) [inline src/HttpCacheListener.php]: Re-instate the `$priority = 1` argument for forwards-compatibility purposes (v3 adds it to the interface).

**Comment by weierophinney** [comment]
Two things:

- Check the travis-ci output to see what the failures are.
- In `Module.php`, can you change line [30](https://github.com/zfcampus/zf-http-cache/blob/master/Module.php#L30) to read:
  ```php
  $sm->get('ZF\HttpCache\HttpCacheListener')->attach($em);
  ```
  This will prepare it for v3 of the event manager, making the new version we tag from this forwards compatible.

Thanks!

**Comment by corentin-larose** [comment]
- .travis.yml outdated (bad php versions) => fixed
- added forward compat

File: Module.php [hunk 2]
```diff
@@ -8,9 +8,9 @@ class Module
      */
     public function getAutoloaderConfig()
     {
-        return array('Zend\Loader\StandardAutoloader' => array('namespaces' => array(
+        return [\Zend\Loader\StandardAutoloader::class => ['namespaces' => [
             __NAMESPACE__ => __DIR__.'/src/',
```
Comment (weierophinney) [inline Module.php]: Ugh; need to update the `.php_cs` settings to remove the "remove spaces around concatenation operators" rule. I can do this on merge, though.

Comment (weierophinney) [inline Module.php]: However: *do* import the StandardAutoloader class, instead of using global qualification.

File: Module.php [hunk 3]
```diff
@@ -27,6 +27,6 @@ public function onBootstrap(\Zend\Mvc\MvcEvent $e)
         $em  = $app->getEventManager();
         $sm  = $app->getServiceManager();
 
-        $em->attachAggregate($sm->get('ZF\HttpCache\HttpCacheListener'));
+        $sm->get(\ZF\HttpCache\HttpCacheListener::class)->attach($em);
     }
```
Comment (weierophinney) [inline Module.php]: You're already in the `ZF\HttpCache` namespace, so you don't need to globally qualify this. :smile:

File: config/module.config.php [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-return array(
-    'service_manager' => array(
-        'factories' => array(
-            'ZF\HttpCache\HttpCacheListener' => 'ZF\HttpCache\HttpCacheListenerFactory',
-        ),
-    ),
+return [
+    'service_manager' => [
+        'factories' => [
+            \ZF\HttpCache\HttpCacheListener::class => \ZF\HttpCache\HttpCacheListenerFactory::class,
```
Comment (weierophinney) [inline config/module.config.php]: To make things more readable, I often import such classes at the top of the file.

File: test/HttpCacheListenerTest.php [hunk 5]
```diff
@@ -509,9 +509,9 @@ public function testCheckStatusCode(array $config, $code, $exResult)
      */
     public function testMethodsReturnsSelf($method, $args)
     {
-        $ret = call_user_func_array(array($this->instance, $method), $args);
+        $ret = call_user_func_array([$this->instance, $method], $args);
 
-        $this->assertInstanceOf('\ZF\HttpCache\HttpCacheListener', $ret);
+        $this->assertInstanceOf(\ZF\HttpCache\HttpCacheListener::class, $ret);
```
Comment (weierophinney) [inline test/HttpCacheListenerTest.php]: Class is already imported; no need to globally qualify.

**Pull request closed without merging**